samples/cxl_host_exerciser/dfl-he-cache.h

```diff
@@ -0,0 +1,133 @@
+ * Caller provides struct dfl_fpga_port_region_info with index value set.
+ * Driver returns the region info in other fields.
+ * Return: 0 on success, -errno on failure.
+ */
+struct dfl_he_cache_region_info {
+  /* Input */
+  __u32 argsz; /* Structure length */
+  /* Output */
+  __u32 flags;                             /* Access permission */
+#define DFL_HE_CACHE_REGION_READ (1 << 0)  /* Region is readable */
+#define DFL_HE_CACHE_REGION_WRITE (1 << 1) /* Region is writable */
+#define DFL_HE_CACHE_REGION_MMAP (1 << 2)  /* Can be mmaped to userspace */
+  __u64 size;                              /* Region size (bytes) */
+  __u64 offset; /* Region offset from start of device fd */
+};
+
+#define DFL_HE_CACHE_SET_DSM_INFO _IO(DFL_HE_CACHE_MAGIC, DFL_HE_CACHE_BASE + 3)
+
+struct dfl_he_cache_dsm_info {
+  /* Input */
+  __u32 argsz;     /* Structure length */
+  __u64 user_addr; /* Process virtual address */
+  __u64 length;    /* Length of mapping (bytes)*/
+};
+
+#define DFL_HE_CACHE_CLEAR_DSM_INFO                                            \
+  _IO(DFL_HE_CACHE_MAGIC, DFL_HE_CACHE_BASE + 4)
+
+#define DFL_HE_CACHE_ALLOC_ADDR_TABLE                                          \
+  _IO(DFL_HE_CACHE_MAGIC, DFL_HE_CACHE_BASE + 5)
+#define DFL_HE_CACHE_FREE_ADDR_TABLE                                           \
+  _IO(DFL_HE_CACHE_MAGIC, DFL_HE_CACHE_BASE + 6)
+#define DFL_HE_CACHE_APPEND_ADDR_TABLE                                         \
+  _IO(DFL_HE_CACHE_MAGIC, DFL_HE_CACHE_BASE + 7)
+
+#define DFL_HE_CACHE_NUM_LINES_MIN 1
+#define DFL_HE_CACHE_NUM_LINES_MAX 0xffff
+
+struct dfl_he_cache_addr_table {
+  /* Input */
+  __u32 argsz; /* Structure length */
+  __u32 flags; /* Address Table ID */
+#define DFL_HE_CACHE_READ_ADDR_TABLE (1 << 0)
+#define DFL_HE_CACHE_WRITE_ADDR_TABLE (1 << 1)
+  __u32 cache_lines; /* Buffer size/offset in cache lines */
+};
+
+#define DFL_HE_CACHE_NUMA_DMA_MAP                                              \
+  _IO(DFL_HE_CACHE_MAGIC, DFL_HE_CACHE_BASE + 12)
+#define DFL_HE_CACHE_NUMA_DMA_UNMAP                                            \
+  _IO(DFL_HE_CACHE_MAGIC, DFL_HE_CACHE_BASE + 13)
+
+#define DFL_ARRAY_MAX_SIZE 0x10
+
+struct dfl_he_cache_dma_map {
+  /* Input */
+  __u32 argsz;                         /* Structure length */
+  __u32 flags;                         /* flags */
+  __u64 user_addr;                     /* Process virtual address */
+  __u64 length;                        /* Length of mapping (bytes)*/
+  __u32 numa_node;                     /* Node 0,1 2 */
+  __u64 csr_array[DFL_ARRAY_MAX_SIZE]; /* CSR  */
+};
+
+struct dfl_he_cache_dma_unmap {
+  /* Input */
+  __u32 argsz;                         /* Structure length */
+  __u32 flags;                         /* flags */
+  __u64 user_addr;                     /* Process virtual address */
+  __u64 length;                        /* Length of mapping (bytes)*/
+  __u64 csr_array[DFL_ARRAY_MAX_SIZE]; /* CSR  */
```

Comment:
Please remove this.  We discussed that it would make the driver difficult to upstream.